File: src/main/kotlin/de/thm/mni/ii/classroom/event/UserEvent.kt
package de.thm.mni.ii.classroom.event

import com.fasterxml.jackson.annotation.JsonIgnore
import de.thm.mni.ii.classroom.model.classroom.User

data class UserEvent(
    val user: User,
    val userAction: UserAction
) : ClassroomEvent(UserEvent::class.simpleName!!) {
    @JsonIgnore override fun getClassroomId(): String = this.user.classroomId
}

enum class UserAction {
    JOIN,
    VISIBILITY_CHANGE,
    LEAVE
}

File: web-gui/src/app/rsocket/event/UserEvent.ts
import {User} from "../../model/User";
import {ClassroomEvent} from "./ClassroomEvent";

export class UserEvent extends ClassroomEvent {
  user: User
  userAction: UserAction

  constructor() {
      super("UserEvent");
  }
}

export enum UserAction {
  JOIN = "JOIN",
  LEAVE = "LEAVE",
  VISIBILITY_CHANGE = "VISIBILITY_CHANGE",
}
